File: demos/demo1/weblinks-ios/ViewController.m
//
//  ViewController.m
//  weblinks-ios
//
//  Created by Dominik Pich on 04/01/15.
//  Copyright (c) 2015 Dominik Pich. All rights reserved.
//

#import "ViewController.h"
#import <weblinks/weblinks.h>

void myGenericErrorFunc(id null, const char *msg, ...);
void myGenericErrorFunc(id null, const char *msg, ...)
{
    va_list vargs;
    va_start(vargs, msg);
    
    NSString *format = [NSString stringWithUTF8String:msg];
    NSMutableString *str = [[NSMutableString alloc] initWithFormat:format arguments:vargs];
    
    NSLog(@"%@", str);
    
    va_end(vargs);
}

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    UITextView *textView = self.view.subviews[0];
    
    @autoreleasepool {
        NSString *path = [[NSBundle mainBundle] pathForResource:@"weblinks" ofType:@"xml"];
        WLFG *fg = [WLFG FGFromFile:path];
        if(fg)
            textView.text = fg.dictionary;
    }
}

@end

File: demos/weblinks-demo/weblinks-ios/ViewController.swift
//
//  ViewController.swift
//  addressTest_ios
//
//  Created by Dominik Pich on 10/01/15.
//  Copyright (c) 2015 Dominik Pich. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var textView: UITextView!

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let url = NSBundle.mainBundle().URLForResource("weblinks", withExtension: "xml")
        if(url != nil) {
            let addr = WLFG.FGFromURL(url!)
            if(addr != nil) {
                textView.text = addr!.dictionary.description
            }
        }
    }


}

